db/cassandra/src/main/java/org/commonjava/indy/cassandra/data/CassandraStoreDataManager.java

```diff
@@ -173,6 +186,140 @@ protected ArtifactStore putArtifactStoreInternal( StoreKey storeKey, ArtifactSto
         return toArtifactStore( dtxArtifactStore );
     }
 
+    @Override
+    public Set<Group> affectedBy( Collection<StoreKey> keys ) throws IndyDataException
+    {
+
+        final Set<Group> result = new HashSet<>();
+
+        // use these to avoid recursion
+        final Set<StoreKey> processed = new HashSet<>();
+        final LinkedList<StoreKey> toProcess = new LinkedList<>( keys );
+
+        while ( !toProcess.isEmpty() )
+        {
+            StoreKey key = toProcess.removeFirst();
+            if ( key == null )
+            {
+                continue;
+            }
+
+            if ( processed.add( key ) )
+            {
+                DtxAffectedStore affectedStore = storeQuery.getAffectedStore( key );
+                Set<StoreKey> affected = affectedStore.getAffectedStoreKeys();
+                if ( affected != null )
+                {
+                    logger.debug( "Get affectedByStores, key: {}, affected: {}", key, affected );
+                    affected = affected.stream().filter( k -> k.getType() == group ).collect( Collectors.toSet() );
+                    for ( StoreKey gKey : affected )
+                    {
+                        // avoid loading the ArtifactStore instance again and again
+                        if ( !processed.contains( gKey ) && !toProcess.contains( gKey ) )
+                        {
+                            ArtifactStore store = getArtifactStoreInternal( gKey );
+
+                            // if this group is disabled, we don't want to keep loading it again and again.
+                            if ( store.isDisabled() )
+                            {
+                                processed.add( gKey );
```

Comment:
Should we also filter the disabled groups out from the result here? Seems this logic shows only enabled groups will be added to final result.

Reply:
Yes, it returns only enabled ones from the code. We need to refactor the method as following to support both. 
```
Set<Group> getGroupsAffectedBy( Collection<StoreKey> keys )
        throws IndyDataException;

Set<Group> getGroupsAffectedBy( Collection<StoreKey> keys, Boolean enabled )
                throws IndyDataException;
```